test/dynamic_config_test.go

```diff
@@ -0,0 +1,264 @@
+						Expect(ok).ToNot(BeFalse())
+
+						Expect(reportDataOp).To(Equal("test"))
+
+						By("Verify no warm/cold restarts in Pods")
+						validateServerRestart(ctx, aeroCluster, podPIDMap, false)
+
+						By("Modify dynamic config by removing fields")
+						aeroCluster, err = getCluster(
+							k8sClient, ctx, clusterNamespacedName,
+						)
+						Expect(err).ToNot(HaveOccurred())
+
+						podPIDMap, err = getPodIDs(ctx, aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						delete(aeroCluster.Spec.AerospikeConfig.Value["service"].(map[string]interface{}), "proto-fd-max")
+						delete(aeroCluster.Spec.AerospikeConfig.Value["security"].(map[string]interface{}), "log")
+
+						err = updateCluster(k8sClient, ctx, aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						pod = aeroCluster.Status.Pods["dynamic-config-test-0-0"]
+
+						conf, err = getAerospikeConfigFromNode(logger, k8sClient, ctx, clusterNamespacedName,
+							"service", &pod)
+						Expect(err).ToNot(HaveOccurred())
+						cv, ok = conf["proto-fd-max"]
+						Expect(ok).ToNot(BeFalse())
+
+						Expect(cv).To(Equal(int64(15000)))
+
+						conf, err = getAerospikeConfigFromNode(logger, k8sClient, ctx, clusterNamespacedName,
+							"security", &pod)
+						Expect(err).ToNot(HaveOccurred())
+
+						_, ok = conf["log.report-data-op[0]"].(string)
+						Expect(ok).ToNot(BeTrue())
+
+						By("Verify no warm/cold restarts in Pods")
+						validateServerRestart(ctx, aeroCluster, podPIDMap, false)
+
+					},
+				)
+
+				It(
+					"Should update config statically", func() {
+
+						By("Modify static config")
+
+						aeroCluster, err := getCluster(
+							k8sClient, ctx, clusterNamespacedName,
+						)
+						Expect(err).ToNot(HaveOccurred())
+
+						podPIDMap, err := getPodIDs(ctx, aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						aeroCluster.Spec.AerospikeConfig.Value["security"].(map[string]interface{})["enable-quotas"] = false
+						dc := map[string]interface{}{
+							"name":      "dc2",
+							"auth-mode": "internal",
+							"auth-user": "admin",
+							"node-address-ports": []string{
+								"aeroclusterdst-0-0 3000",
+							},
+							"auth-password-file": "/etc/aerospike/secret/password_DC1.txt",
+							"namespaces": []map[string]interface{}{
+								{
+									"name": "test",
+								},
+							},
+						}
+
+						aeroCluster.Spec.AerospikeConfig.Value["xdr"].(map[string]interface{})["dcs"] = append(
+							aeroCluster.Spec.AerospikeConfig.Value["xdr"].(map[string]interface{})["dcs"].([]interface{}), dc)
+
+						err = updateCluster(k8sClient, ctx, aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						pod := aeroCluster.Status.Pods["dynamic-config-test-0-0"]
+
+						conf, err := getAerospikeConfigFromNode(logger, k8sClient, ctx, clusterNamespacedName,
+							"security", &pod)
+						Expect(err).ToNot(HaveOccurred())
+
+						enableQuotas, ok := conf["enable-quotas"].(bool)
+						Expect(ok).ToNot(BeFalse())
```

Comment:
This seems redundant as we are fetching info from node and the default value of that field itself is false.
I think its better to set it to true before update or use different field for testing